cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/condition/BasicConditionContext.java

```diff
@@ -74,6 +142,6 @@ public void execute(int timeoutInSeconds, TxRunnable runnable) throws Transactio
 
   @Override
   public Map<String, StageStatistics> getStageStatistics() {
-    return new HashMap<>();
+    return ImmutableMap.copyOf(stageStatistics);
```

Comment:
you can make it immutable in the constructor instead of here, to prevent copying every time somebody calls this.